packages/salesforcedx-vscode-core/src/commands/templates/forceFunctionCreate.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  CancelResponse,
+  ContinueResponse,
+  FunctionInfo,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { nls } from '../../messages';
+import {
+  CompositeParametersGatherer,
+  SfdxCommandlet,
+  SfdxWorkspaceChecker
+} from '../util';
+import { BaseTemplateCommand } from './baseTemplateCommand';
+import { FUNCTION_TYPE } from './metadataTypeConstants';
+
+export class ForceFunctionCreateExecutor extends BaseTemplateCommand {
+  constructor() {
+    super(FUNCTION_TYPE);
+  }
+
+  public build(data: FunctionInfo): Command {
+    if (data.language === 'javascript') {
+      this.setFileExtension('js');
+    } else if (data.language === 'typescript') {
+      this.setFileExtension('ts');
+    }
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_function_create_text'))
+      .withArg('evergreen:function:create')
+      .withArg(data.fileName)
+      .withFlag('--language', data.language)
+      .withLogName('force_org_create_function')
+      .build();
+  }
+}
+
+export class FunctionInfoGatherer implements ParametersGatherer<FunctionInfo> {
+  public async gather(): Promise<CancelResponse | ContinueResponse<FunctionInfo>> {
+    const nameInputOptions = {
+      prompt: nls.localize('force_function_enter_function')
+    } as vscode.InputBoxOptions;
+    const name = await vscode.window.showInputBox(nameInputOptions);
+    if (name === undefined) {
+      return { type: 'CANCEL' };
+    }
+
+    const language = await vscode.window.showQuickPick(['javascript', 'typescript'], {
+      placeHolder: nls.localize('force_function_enter_language')
+    });
+
+    if (language === undefined) {
+      return { type: 'CANCEL' };
+    }
+    return {
+      type: 'CONTINUE',
+      data: {
+        fileName: name,
+        language,
+        outputdir: ''
```

Comment:
will output dir be used? If not let's update `FunctionInfo` to just include `fileName` and `language`

Reply:
Agree! that was my first thought too. But that required far more refactoring since every other template has outputdir and extends DirFileNameSelection (that has outputdir) and by not doing that, I wouldn't be able to use all the template related existing code. I can add a comment here why we are doing this here. Happy to discuss on hangout if there are other options.

Reply:
Yeah, I agree. Thanks for providing this background. It would be helpful to have a comment though. IMHO strategy pattern and the way our base class BaseTemplateCommand is structured are a bit overengineering for the purpose of processing output folder and file names - Let's not worry about it now.